glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -97,12 +97,47 @@ open class GleanInternalAPI internal constructor () {
         }
     }
 
+    /**
+     * Enable or disable Glean collection and upload.
+     *
+     * Metric collection is enabled by default.
+     *
+     * When uploading is disabled, metrics aren't recorded at all and no data
+     * is uploaded.
+     *
+     * When disabling, all pending metrics, events and queued pings are cleared.
+     *
+     * When enabling, the core Glean metrics are recreated.
+     *
+     * @param enabled When true, enable metric collection.
+     */
+    fun setUploadEnabled(enabled: Boolean) {
```

Comment:
We already expose `glean_set_upload_enabled`, though it doesn't re-do core metrics, etc.